warehouse/query-core/src/main/java/datawave/query/predicate/EventDataQueryFieldFilter.java

```diff
@@ -120,49 +143,43 @@ public boolean peek(@Nullable Map.Entry<Key,String> input) {
      */
     @Override
     public Range getSeekRange(Key current, Key endKey, boolean endKeyInclusive) {
-        // not yet implemented
-        return null;
+        if (current == null || maxNextCount == -1 || nextCount < maxNextCount) {
+            return null;
+        }
+
+        parser.parse(current);
+        String higher = fields.higher(parser.getField());
+
+        Text columnQualifier;
+        if (higher == null) {
+            // generate a rollover range
+            Text columnFamily = new Text(current.getColumnFamilyData().toString() + '\u0000');
+            Key start = new Key(current.getRow(), columnFamily);
+            return new Range(start, false, endKey, endKeyInclusive);
+        } else {
+            // seek to next available field
+            columnQualifier = new Text(higher + '\u0000');
+            Key start = new Key(current.getRow(), current.getColumnFamily(), columnQualifier);
+            return new Range(start, false, endKey, endKeyInclusive);
+        }
     }
 
     @Override
     public int getMaxNextCount() {
-        // not yet implemented
+        // while technically implemented, do not return the max next count here. This method is only used
+        // by the ChainableEventDataQueryFilter which does NOT guarantee that the filter will exclusively
+        // be applied to event keys.
         return -1;
```

Comment:
should this maybe throw an UnsupportedMethodException then?

Reply:
That would be one way to ensure the filter is being used appropriately...I like  it.